### PR TITLE
#984 - New v6 job REST APIs 

### PR DESCRIPTION
**Pull request by mikenholt**
Removed from `Job` model: 
   - `cpus_required` 
   - `mem_required` 
   - `disk_out_required` 
   - `inputs`
   - `outputs`
   - `job_exes`
   - `recipes`

Renamed in `Job` model:  
   - `disk_in_required` to `input_file_size`
   - `data` to `input`
   - `results` to `output`

Added to `Job` model:
   - `resources` section that contains the resources JSON.
   - If the job status is QUEUED, RUNNING, COMPLETED, or FAILED, add an `execution` section that contains the job execution details (like from #983#) where job.num_exes == job_exe.exe_num.
   - Add a `recipe` section that contains the non-superseded recipe (if exists) that matches the job. Use the RecipeBaseSerializer.

### CODE REVIEW

File: scale/job/migrations/0034_auto_20180112_1759.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    operations = [
+        migrations.RenameField(
+            model_name='job',
+            old_name='cpus_required',
+            new_name='input_file_size',
+        ),
+        migrations.RemoveField(
+            model_name='job',
+            name='data',
+        ),
```
Comment (dfaller) [inline scale/job/migrations/0034_auto_20180112_1759.py]: It should be `disk_in_required` being renamed to `input_file_size`.
Also `data` and `results` need to be renamed to `input` and `output`. This migration will drop these columns losing all job input and output information.

File: scale/job/models.py [hunk 2]
```diff
@@ -217,9 +217,50 @@ def get_details(self, job_id):
             'superseded_by_job', 'superseded_by_job__job_type'
         ).get(pk=job_id)
 
+        # Attempt to get related job executions
+        if job.status in ['RUNNING', 'COMPLETED', 'FAILED', 'QUEUED']:
+            job_exe = JobExecution.objects.filter(job=job, exe_num=job.num_exes).select_related('job', 'node', 'error')
+            job.execution = job_exe.defer('job__input', 'job__output').order_by('-created')
+        else:
+            job.execution = []
```
Comment (dfaller) [inline scale/job/models.py]: We should reuse the `JobExecutionManager.get_job_exe_details()` method here.
Also update that method to select related on `jobexecutionend__error` and defer `job__input` and `job__output`.

We should use a try/except block on `JobExecution.DoesNotExist` (just like the execution detail view) and set `job.execution = None` if there is no execution.

File: scale/job/models.py [hunk 3]
```diff
@@ -1032,21 +1061,30 @@ def get_resources(self):
...
-        output_size_mb = long(math.ceil(multiplier * disk_in_required + const))
+        output_size_mb = long(math.ceil(multiplier * input_file_size + const))
         disk_out_required = max(output_size_mb, MIN_DISK)
 
-        resources.add(NodeResources([Mem(memory_required), Disk(disk_out_required + disk_in_required)]))
+        resources.add(NodeResources([Mem(memory_required), Disk(disk_out_required + input_file_size)]))
         return resources
 
+    def get_resources_dict(self):
+        """Gathers resources information and returns it as a dictl.
```
Comment (dfaller) [inline scale/job/models.py]: Minor typo

File: scale/job/models.py [hunk 4]
```diff
@@ -219,10 +219,12 @@ def get_details(self, job_id):
 
         # Attempt to get related job executions
         if job.status in ['RUNNING', 'COMPLETED', 'FAILED', 'QUEUED']:
-            job_exe = JobExecution.objects.filter(job=job, exe_num=job.num_exes).select_related('job', 'node', 'error')
-            job.execution = job_exe.defer('job__input', 'job__output').order_by('-created')
+            try:
+                job.execution = JobExecution.objects.get_job_exe_details(job_id=job.id, exe_num=job.num_exes)
+            except JobExecution.DoesNotExist():
```
Comment (dfaller) [inline scale/job/models.py]: Last thing: extra parens here. Wait to merge until after the release is done. Well done.

**Comment by mikenholt** [comment]
BINGO BANGO

**Pull request merged**